compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -179,18 +179,11 @@ impl<'tcx> MirSource<'tcx> {
 
 #[derive(Clone, TyEncodable, TyDecodable, Debug, HashStable, TypeFoldable, TypeVisitable)]
 pub struct GeneratorInfo<'tcx> {
-    /// The yield type of the function, if it is a generator.
-    pub yield_ty: Option<Ty<'tcx>>,
-
     /// Generator drop glue.
-    pub generator_drop: Option<Body<'tcx>>,
+    pub generator_drop: Body<'tcx>,
 
     /// The layout of a generator. Produced by the state transformation.
-    pub generator_layout: Option<GeneratorLayout<'tcx>>,
-
-    /// If this is a generator then record the type of source expression that caused this generator
-    /// to be created.
-    pub generator_kind: GeneratorKind,
+    pub generator_layout: GeneratorLayout<'tcx>,
```

Comment:
Same here. I'm guessing it's because previously these would be `None` before we figured out what the generator layout was, but now you don't get the `GeneratorInfo` until after it's been computed?

Reply:
The difference is that the `Option` is moved around `GeneratorInfo`: we only create the struct when we know this is a generator.